File: pmd/java/aulas/0408/Complexo.java
public class Complexo {

  private double a, b;

  public Complexo() {
    this.a = this.b = 0;
  }

  public Complexo(double a, double b) {
    this.a = a;
    this.b = b;
  }

  public Complexo soma(Complexo c) {
    return new Complexo(this.a + c.a, this.b + c.b);
  }

  public Complexo produto(Complexo c) {
    return new Complexo(
        this.a * c.a - this.b * c.b,
        this.b * c.a + this.a * c.b);
  }

  public Complexo conjugado() {
    return new Complexo(this.a, -this.b);
  }

  public String toString() {
    // return "(" + this.a + " + " + this.b + "i)";
    return String.format("(%.2f + %.2fi)", this.a, this.b);
  }

  public static void main(String[] args) {
    Complexo z = new Complexo(2, 3);
    Complexo w = new Complexo(4, 5);
    System.out.println(z + " + " + w + " = " + z.soma(w));

  }

}

File: pmd/python/aulas/complexo.py
class Complexo:

  def __init__(self, a=0, b=0):
    self.a = a
    self.b = b
  
  def soma(self, c):
    return Complexo(self.a + c.a, self.b + c.b)

  def produto(self, c):
    return Complexo(
      self.a * c.a - self.b * c.b,
      self.b * c.a + self.a * c.b)

  def conjugado(self):
    return Complexo(self.a, -self.b)
  

  def __str__(self):
    return f"({self.a} + {self.b}i)"


#main
z = Complexo(2, 3)
w = Complexo(4, 5)
print(f"{z} + {w} = {z.soma(w)}")